docs/doc_check/CMakeLists.txt

```diff
@@ -1,5 +1,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
+if (NOT ONNX_MLIR_BUILD_TESTS)
+  set(EXCLUDE_FROM_ALL ON)
+endif()
```

Comment:
I don't think this does anything. `EXCLUDE_FROM_ALL` is a target property. Setting a variable called `EXCLUDE_FROM_ALL` won't do anything. The proper use of `EXCLUDE_FROM_ALL` should be like the code in `MLIR.cmake`:
```
  if (EXCLUDE_FROM_ALL)
    add_executable(${name} EXCLUDE_FROM_ALL ${ARG_UNPARSED_ARGUMENTS})
  else()
    add_executable(${name} ${ARG_UNPARSED_ARGUMENTS})
  endif()
```
The variable `EXCLUDE_FROM_ALL` is used to decide whether to add the property `EXCLUDE_FROM_ALL` to a target.

Reply:
@gongsu832 I reused this pattern from the current `onnx-mlir`. I know it does not make it right, but on my Mac it just seems to work. When I set the variable as `ON`, it builds all the tests as before. When I set it to off, the resulting build does not recognize the `make check-onyx-backend` as a valid target.

Let me run the experiment on LoZ too.

Reply:
I assume you meant `check-onnx-backend` since there is no `check-onyx-backend`.

Regardless of how the variable is set, `check-onnx-backend` should always be a valid target. The difference is whether it's included in the default `all` target.

Reply:
Also note that `add_executable`, `add_library`, etc. behave differently from `add_custom_target`. By default, `add_executable`, `add_library`, etc. will add the target to `all` unless you specify `EXCLUDE_FROM_ALL` on the target. `add_custom_target`, on the other hand, by default will not add the target to `all` unless you specify `ALL` on the target.

Reply:
@gongsu832 Are there specific steps you would like me to take? Or if it takes you less time, do you want to take over this PR to do it the right way (if the current approach is not the preferred approach).

Reply:
I can do it but probably won't be able to get to it until some time next week.

Reply:
Thanks, there is no rush, glad it will be done using the proper approach